### PR TITLE
compose: enable entrypoint as stringArray

### PR DESCRIPTION
**Pull request by fahedouch**
I am raising this fix to handle this case https://github.com/containerd/nerdctl/discussions/1069 
Signed-off-by: fahed dorgaa <fahed.dorgaa@gmail.com>

### CODE REVIEW

File: cmd/nerdctl/run.go [hunk 1]
```diff
@@ -693,7 +694,11 @@ func generateRootfsOpts(ctx context.Context, client *containerd.Client, platform
 		}
 		var processArgs []string
 		if entrypoint != "" {
-			processArgs = append(processArgs, entrypoint)
+			s, err := shellwords.Parse(entrypoint)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			processArgs = append(processArgs, s...)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/run.go:701]: Seems incompatible with Docker.
Docker doesn't parse the entrypoint.
```
$ docker run --rm --entrypoint "echo foo bar" alpine
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "echo foo bar": executable file not found in $PATH: unknown.
```

Comment (AkihiroSuda) [reply cmd/nerdctl/run.go:701]: I think we should just change the type of the `entrypoint` from `string` to `[]string`.

Comment (fahedouch) [reply cmd/nerdctl/run.go:701]: @AkihiroSuda I am rethinking about it , switching `entrypoint` to slice this means sthg like this 
`docker run --rm  --entrypoint "foo" --entrypoint "echo"  alpine` which is confusing.
FYI `docker-compose --entrypoint` is parsing shell words.
`nerdctl compose` rely on `nerdctl` which is not the same behavior for docker-compose and docker. Keeping 100% compatibility  is tough in this situation

Comment (AkihiroSuda) [reply cmd/nerdctl/run.go:701]: > docker run --rm --entrypoint "foo" --entrypoint "echo" alpine which is confusing.

I think this is fine as an internal implementation detail of `nerdctl compose`.
`nerdctl run` users are not expected to specify multiple `--entrypoint` flags manually.

Comment (fahedouch) [reply cmd/nerdctl/run.go]: fixed

File: pkg/composer/serviceparser/serviceparser.go [hunk 2]
```diff
@@ -478,12 +478,6 @@ func newContainer(project *compose.Project, parsed *Service, i int) (*Container,
 	for _, v := range svc.DNS {
 		c.RunArgs = append(c.RunArgs, fmt.Sprintf("--dns=%s", v))
 	}
-
-	if len(svc.Entrypoint) > 1 {
-		return nil, fmt.Errorf("service %s: specifying entrypoint with multiple strings (%v) is not supported yet",
-			svc.Name, svc.Entrypoint)
-	}
```
Comment (AkihiroSuda) [inline pkg/composer/serviceparser/serviceparser.go:485]: Could you also update https://github.com/containerd/nerdctl/blob/master/docs/compose.md#incompatibility ?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: pkg/composer/serviceparser/serviceparser.go (hunk 2 above)
Comment (AkihiroSuda) [reply pkg/composer/serviceparser/serviceparser.go:485]: Pushed a commit by myself https://github.com/containerd/nerdctl/pull/1070/commits/08924b9c54f4df48ce02090ad6c05f26c1bf4aa1

**Pull request merged**